tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -127,163 +128,207 @@ public void resolve() {
     typeMap = inferrer.getTypeMap();
   }
 
-  public DAGRequest buildScan(boolean idxScan) {
-    if (idxScan) {
-      return buildIndexScan();
-    } else {
-      return buildTableScan();
-    }
+  private ColumnInfo handleColumn() {
```

Comment:
Isn't this an immutable one? Can make it static final instead of generate one each time. This also is better for reading since it indicate the field is an constant static unchanged placement.

Reply:
done